File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int len = s.size();
        if (len <= 1) return len;

        set<char> st;
        int maxx = 0, i = 0;  

        for (int j = 0; j < len; j++) {  
            while (st.find(s[j]) != st.end()) {
                st.erase(s[i]);
                i++;  
            }
            st.insert(s[j]);
            maxx = max(maxx, j - i + 1);
        }

        return maxx;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int len = s.length();
        if (len <= 1) return len;

        HashSet<Character> st = new HashSet<>();
        int maxx = 0, i = 0;  

        for (int j = 0; j < len; j++) {  
             while (st.contains(s.charAt(j))) {
                st.remove(s.charAt(i));
                i++; 
            }
            st.add(s.charAt(j));
            maxx = Math.max(maxx, j - i + 1);
        }

        return maxx;
    }
}